### PR TITLE
feat(sendHtml/legacy): only check user agent for ie

### PR DESCRIPTION
**Pull request by JAdshead**
## Description

Removed `browserslist-useragent` and implemented simple check for IE user agent.

## Motivation and Context

This has both performance and security improvements. 
`browserslist-useragent` uses `useragent` to parse the user agent which in turn uses a huge list of [regular expressions](https://github.com/3rd-Eden/useragent/blob/master/lib/regexps.js) to parse the user agent string. This is both costly in performance and could potentially be introducing [security vulnerabilities](https://medium.com/@nitinpatel_20236/what-are-evil-regexes-7b21058c747e). 

## How Has This Been Tested?

In addition to unit and integration tests the One App server was started with `frank-llyod-root. autocannon was used to hit `localhost:3000/success` with 50 connections for a duration of 30 seconds.  

`autocannon localhost:3000/success -c 50 -d 30 --headers "Accept-Language= en-USen;q=0.5;" --headers "User-Agent= curl/7.5.0"` 

This was run three times before and after the changes. The fastest results are posted below

Before:

```
┌─────────┬───────┬───────┬───────┬────────┬──────────┬──────────┬───────────┐
│ Stat    │ 2.5%  │ 50%   │ 97.5% │ 99%    │ Avg      │ Stdev    │ Max       │
├─────────┼───────┼───────┼───────┼────────┼──────────┼──────────┼───────────┤
│ Latency │ 60 ms │ 64 ms │ 85 ms │ 126 ms │ 66.29 ms │ 10.08 ms │ 181.62 ms │
└─────────┴───────┴───────┴───────┴────────┴──────────┴──────────┴───────────┘
┌───────────┬─────────┬─────────┬─────────┬────────┬─────────┬────────┬─────────┐
│ Stat      │ 1%      │ 2.5%    │ 50%     │ 97.5%  │ Avg     │ Stdev  │ Min     │
├───────────┼─────────┼─────────┼─────────┼────────┼─────────┼────────┼─────────┤
│ Req/Sec   │ 651     │ 651     │ 759     │ 787    │ 748     │ 38.47  │ 651     │
├───────────┼─────────┼─────────┼─────────┼────────┼─────────┼────────┼─────────┤
│ Bytes/Sec │ 3.64 MB │ 3.64 MB │ 4.24 MB │ 4.4 MB │ 4.18 MB │ 215 kB │ 3.63 MB │
└───────────┴─────────┴─────────┴─────────┴────────┴─────────┴────────┴─────────┘

Req/Bytes counts sampled once per second.

22k requests in 30.06s, 125 MB read
```

After:

```
┌─────────┬───────┬───────┬───────┬────────┬─────────┬─────────┬───────────┐
│ Stat    │ 2.5%  │ 50%   │ 97.5% │ 99%    │ Avg     │ Stdev   │ Max       │
├─────────┼───────┼───────┼───────┼────────┼─────────┼─────────┼───────────┤
│ Latency │ 49 ms │ 51 ms │ 64 ms │ 102 ms │ 52.9 ms │ 7.47 ms │ 169.98 ms │
└─────────┴───────┴───────┴───────┴────────┴─────────┴─────────┴───────────┘
┌───────────┬─────────┬─────────┬─────────┬────────┬─────────┬────────┬─────────┐
│ Stat      │ 1%      │ 2.5%    │ 50%     │ 97.5%  │ Avg     │ Stdev  │ Min     │
├───────────┼─────────┼─────────┼─────────┼────────┼─────────┼────────┼─────────┤
│ Req/Sec   │ 870     │ 870     │ 940     │ 964    │ 935.64  │ 21.66  │ 870     │
├───────────┼─────────┼─────────┼─────────┼────────┼─────────┼────────┼─────────┤
│ Bytes/Sec │ 4.79 MB │ 4.79 MB │ 5.17 MB │ 5.3 MB │ 5.15 MB │ 119 kB │ 4.79 MB │
└───────────┴─────────┴─────────┴─────────┴────────┴─────────┴────────┴─────────┘

Req/Bytes counts sampled once per second.

28k requests in 30.06s, 154 MB read
```
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [x] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

None

### CODE REVIEW

File: src/server/middleware/sendHtml.js [hunk 1]
```diff
@@ -33,6 +31,16 @@ const { buildVersion } = readJsonFile('../../../.build-meta.json');
...
 
+// http://www.useragentstring.com/pages/useragentstring.php?name=Internet+Explorer
+const legacyIndicators = [
+  'rv:11', // IE 11 on mobile
+  'MSIE', // IE
+];
+
+const legacyUserAgent = (userAgent) => (
+  legacyIndicators.some((legacyIndicator) => userAgent.indexOf(legacyIndicator) !== -1)
+);
```
[the remark this answers is not in the document]
Comment (JAdshead) [reply src/server/middleware/sendHtml.js:42]: will run some tests

[the remark this answers is not in the document]
Comment (JAdshead) [reply src/server/middleware/sendHtml.js:42]: Seems to be fractionally slower but not enough difference to really call

```
┌─────────┬───────┬───────┬───────┬────────┬──────────┬─────────┬───────────┐
│ Stat    │ 2.5%  │ 50%   │ 97.5% │ 99%    │ Avg      │ Stdev   │ Max       │
├─────────┼───────┼───────┼───────┼────────┼──────────┼─────────┼───────────┤
│ Latency │ 49 ms │ 53 ms │ 66 ms │ 104 ms │ 54.33 ms │ 7.84 ms │ 186.88 ms │
└─────────┴───────┴───────┴───────┴────────┴──────────┴─────────┴───────────┘
┌───────────┬─────────┬─────────┬─────────┬─────────┬─────────┬────────┬─────────┐
│ Stat      │ 1%      │ 2.5%    │ 50%     │ 97.5%   │ Avg     │ Stdev  │ Min     │
├───────────┼─────────┼─────────┼─────────┼─────────┼─────────┼────────┼─────────┤
│ Req/Sec   │ 827     │ 827     │ 920     │ 959     │ 911.2   │ 33.21  │ 827     │
├───────────┼─────────┼─────────┼─────────┼─────────┼─────────┼────────┼─────────┤
│ Bytes/Sec │ 4.55 MB │ 4.55 MB │ 5.06 MB │ 5.28 MB │ 5.01 MB │ 183 kB │ 4.55 MB │
└───────────┴─────────┴─────────┴─────────┴─────────┴─────────┴────────┴─────────┘

Req/Bytes counts sampled once per second.

27k requests in 30.06s, 150 MB read
```

[the remark this answers is not in the document]
Comment (JAdshead) [reply src/server/middleware/sendHtml.js]: 3bc2ef2

**Comment by mtomcal** [comment]
> This change requires cross browser checks.

I think we would want to use Browser Stack or something and check the multiple legacy IE's to make sure legacy bundles are served in the network tab to add confidence.

These results are exciting on the performance end!

**Comment by JAdshead** [comment]
> > This change requires cross browser checks.
> 
> I think we would want to use Browser Stack or something and check the multiple legacy IE's to make sure legacy bundles are served in the network tab to add confidence.
> 
> These results are exciting on the performance end!
I have cross referenced a few resources for all [valid IE user agents](http://www.useragentstring.com/pages/useragentstring.php?name=Internet+Explorer)
and tested making curl requests like the one below to also test the legacy bundles were being returned. 

`curl http://localhost:3000/success  -H "User-Agent: Mozilla/5.0 (compatible, MSIE 11, Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"`

**Pull request merged**